internal/server/server.go

```diff
@@ -266,3 +274,25 @@ func fetchDefaultJobScheduleMapping() map[JobType]string {
 		ExpiringGrantNotification: "0 9 * * *",
 	}
 }
+
+func getAuthInterceptor(config *Config) (grpc.UnaryServerInterceptor, error) {
+	// default fallback to user email on header
+	authInterceptor := withAuthenticatedUserEmail(config.Auth.Default.HeaderKey)
+
+	if config.Auth.Provider == "oidc" {
+		idtokenValidator, err := idtoken.NewValidator(context.Background())
+		if err != nil {
+			return nil, err
+		}
+
+		params := auth.OIDCValidatorParams{
+			Audience:          config.Auth.OIDC.Audience,
+			ValidEmailDomains: config.Auth.OIDC.EligibleEmailDomains,
+		}
+
+		bearerTokenValidator := auth.NewOIDCValidator(idtokenValidator, params)
```

Comment:
```suggestion
		bearerTokenValidator := auth.NewOIDCValidator(idtokenValidator, config.Auth.OIDC)
```

is it possible if we just use oidc config and get rid of `OIDCValidatorParams`?

Reply:
that's right. i forgot why i used that params but i have refactored it anyway 😂